fastlane_core/lib/fastlane_core/update_checker/update_checker.rb

```diff
@@ -108,20 +108,41 @@ def self.ios_app_identifier
       nil # we don't want this method to cause a crash
     end
 
+    # (optional) Returns the app identifier for the current tool
+    # supply and screengrab use different param names so we have to special case here
+    def self.android_app_identifier(gem_name)
+      # ARGV example:
+      # ["-a", "com.krausefx.app"]
+      ARGV.each_with_index do |current, index|
+        if (current == "--package_name" || current == "--app_package_name" || (current == '-p' && gem_name == 'supply') || (current == '-a' && gem_name == 'screengrab') )
```

Comment:
Extract long conditional to a named method? `if app_identifier_arg?(gem_name, current)`